examples/prompt/main.go

```diff
@@ -0,0 +1,23 @@
+package main
+
+import (
+	"strings"
+
+	"github.com/bitfield/script"
+)
+
+func Install(path string) {
+	script.Echo("📂 Installing at " + path + "\n").Stdout()
+	script.Echo("🟢 Successfully installed\n").Stdout()
+}
+
+func main() {
+	script.Echo("Installation of Fake Product\n").Stdout()
+	curDir, err := script.Exec("pwd").String()
+	if err != nil {
+		panic(err)
+	}
+
+	path, _ := script.Echo("Choose install location").ReadInput(strings.TrimSpace(curDir)).String()
```

Comment:
We can use `ExecForEach` here, can't we?

Reply:
I don't really follow, what would `ExecForEach` do here? `ReadInput` reads only one line (reads off `stdin` and stops at newline).

Or do you mean instead of `String()`? Something like:

```diff
diff --git a/examples/read/main.go b/examples/read/main.go
index 55ab3a9..10c31cd 100644
--- a/examples/read/main.go
+++ b/examples/read/main.go
@@ -18,6 +18,6 @@ func main() {
 		panic(err)
 	}
 
-	path, _ := script.Echo("Choose install location").ReadInput(strings.TrimSpace(curDir)).String()
+	path, _ := script.Echo("Choose install location").ReadInput(strings.TrimSpace(curDir)).ExecForEach("echo {{.}}").String()
 	Install(path)
 }
```

I might be misinterpreting your comment, but adding `ExecForEach` just for the sake of using it, might throw off people trying to learn more about `ReadInput`.

Reply:
Well, if we just wanted to read a string as input, we could use `fmt.Scanf`, couldn't we? The value of having it as a pipe method is precisely that we can then _use_ that input in the next stage of the pipe!

In the installer example, we might read the installation path and then pass it to the fictitious `install` command:

```go
script.Echo("install location: ").ReadInput("").ExecForEach("/usr/bin/install {{.}}").Stdout()
```

Reply:
Agreed. I however used `echo Installing at {{.}}` because it has an output. 

When I ran with `/usr/bin/install {{.}}` the example program didn't output anything. I figured having an `echo` makes the example more lively.